src/main/sensors/rpm_filter.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <stdint.h>
+#include <math.h>
+#include "drivers/pwm_output_counts.h"
+#include "sensors/rpm_filter.h"
+#include "common/filter.h"
+#include "sensors/gyro.h"
+#include "build/debug.h"
+#include "flight/mixer.h"
+#include "flight/pid.h"
+
+#define RPM_FILTER_MAXHARMONICS 3
+#define RPM_MOTOR_FILTER_CUTOFF 150
+
+#ifdef USE_RPM_FILTER
+
+static pt1Filter_t rpmFilters[MAX_SUPPORTED_MOTORS];
+
+typedef struct rpmNotchFilter_s 
+{
+    uint8_t harmonics;
+    uint8_t minHz;
+    float   q;
+
+    biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_MAXHARMONICS];
+} rpmNotchFilter_t;
+
+static rpmNotchFilter_t gyroFilter;
+static rpmNotchFilter_t dtermFilter;
+
+static float   erpmToHz;
+static float   filteredMotorErpm[MAX_SUPPORTED_MOTORS];
+static uint8_t numberFilters;
+static uint8_t filterUpdatesPerIteration;
+static float   pidLooptime;
+
+PG_REGISTER_WITH_RESET_FN(rpmFilterConfig_t, rpmFilterConfig, PG_RPM_FILTER_CONFIG, 1);
+
+void pgResetFn_rpmFilterConfig(rpmFilterConfig_t *config) 
+{
+    config->gyro_rpm_notch_harmonics = 3;
+    config->gyro_rpm_notch_min = 100;
+    config->gyro_rpm_notch_q = 50;
+
+    config->dterm_rpm_notch_harmonics = 1;
+    config->dterm_rpm_notch_min = 100;
+    config->dterm_rpm_notch_q = 50;
+}
+
+static void rpmNotchFilterInit(rpmNotchFilter_t* filter, int harmonics, int minHz, int q, float looptime) 
+{
+    filter->harmonics = harmonics;
+    filter->minHz = minHz;
+    filter->q = q / 10.0f;
```

Comment:
A q of 5 is already very high. Since the config is done in ints it seemed appropriate to provide finer granularity than one whole integer for the configuration of Q.